src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx

```diff
@@ -63,124 +66,187 @@ describe("CloneForm", () => {
     state.fabric.loaded = true;
     state.subnet.loaded = true;
     state.vlan.loaded = true;
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <CloneForm
-              clearSidePanelContent={jest.fn()}
-              machines={[]}
-              processingCount={0}
-              viewingDetails={false}
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <CloneForm
+        clearSidePanelContent={jest.fn()}
+        machines={[]}
+        processingCount={0}
+        viewingDetails={false}
+      />,
+      { route: "/machines", state }
     );
-    const isCheckboxDisabled = () =>
-      wrapper.find("input[name='interfaces']").prop("disabled") === true;
-    const isSubmitDisabled = () =>
-      wrapper.find(".p-button--positive[type='submit']").prop("disabled") ===
-      true;
 
     // Checkboxes and submit should be disabled at first.
-    expect(isCheckboxDisabled()).toBe(true);
-    expect(isSubmitDisabled()).toBe(true);
-
-    // Select a source machine - checkbox should be enabled.
-    wrapper.find("[data-testid='machine-select-row']").at(0).simulate("click");
-    await waitForComponentToPaint(wrapper);
-    expect(isCheckboxDisabled()).toBe(false);
-    expect(isSubmitDisabled()).toBe(true);
-
-    // Select config to clone - submit should be enabled.
-    wrapper.find("input[name='interfaces']").simulate("change", {
-      target: { name: "interfaces", value: true },
-    });
-    await waitForComponentToPaint(wrapper);
-    expect(isCheckboxDisabled()).toBe(false);
-    expect(isSubmitDisabled()).toBe(false);
+    expect(
+      screen.getByRole("button", { name: "Clone to machine" })
+    ).toBeDisabled();
+    expect(
+      screen.getByRole("checkbox", { name: "Clone network configuration" })
+    ).toBeDisabled();
+
+    // Select a source machine - form should update
+    await userEvent.click(
+      screen.getByRole("row", { name: machines[1].hostname })
+    );
+
+    expect(
+      screen.getByRole("button", { name: "Clone to machine" })
+    ).toBeDisabled();
+    expect(
+      screen.getByRole("checkbox", { name: "Clone network configuration" })
+    ).not.toBeDisabled();
+
+    // Select config to clone - submit should be re-disabled.
+    await userEvent.click(
+      screen.getByRole("checkbox", { name: "Clone network configuration" })
+    );
+
+    expect(
+      screen.getByRole("checkbox", { name: "Clone network configuration" })
+    ).not.toBeDisabled();
```

Comment:
IMHO it's easier to read (totally optional, not fussed about it).

```suggestion
    ).toBeEnabled();
```

Reply:
I genuinely didn't know about this until now... :facepalm: